pandas/_libs/src/inference.pyx

```diff
@@ -1508,20 +1508,22 @@ def to_object_array(list rows, int min_width=0):
     cdef:
         Py_ssize_t i, j, n, k, tmp
         ndarray[object, ndim=2] result
+        list input_rows
```

Comment:
Is there not a way to make the type declaration part of the call signature? Just wondering if that wouldn't be cleaner. Also curious if doing it in this fashion doesn't require more memory to construct a DataFrame, which maybe is an issue if dealing with very large lists

Reply:
I am not an expert in Cython, but this approach is what I found using Google on their mailing list. :-) So not having it in the signature and cast it later. (Casting still throws an exception if impossible.)

I do not see why would this require any more memory?

Reply:
May not - just not sure how the casting rules are applied. Looking at the module there appear to be other instances where the type is in the signature (see `to_object_array_tuples`). Seems like that would be preferable if its an option

Reply:
What do you mean? The whole issue is that if you put the `list` in the signature, then Cython requires exactly the list signature and not a subclass. So removing the signature and doing casting fixes this. In `to_object_array_tuples` the argument type is `list`, which again means that only strict list type can be provided.

Reply:
I can put `object` type there. It seems equivalent. This is what `infer_dtype` has.